gammapy/estimators/points/profile.py

```diff
@@ -133,12 +133,15 @@ def run(self, datasets):
         profile : `~gammapy.estimators.FluxPoints`
             Profile flux points.
         """
+
         datasets = Datasets(datasets=datasets)
+        datasets_copy = datasets.copy()
+        for dataset in datasets_copy:
+            dataset.background = dataset.npred()
```

Comment:
I have not catched the logics behind...

Reply:
Let's say that you have a point source nearby your extended source of interest. Instead of masking the point source you can include it in the models. The simplest way to do that is to add the npred to the background (because the other source won't be reoptimized as it would be too degenerated in spectrum dataset used internally by the estimator).

Reply:
It can be useful to document this logic in the docstring